Xamarin.Forms.Core/Picker.cs

```diff
@@ -58,5 +78,39 @@ void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
 			SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
 		}
+
+		static void OnItemsSourcePropertyChanged (BindableObject bindable, object value, object newValue)
+		{
+			var picker = (Picker)bindable;
+			var notifyCollection = newValue as INotifyCollectionChanged;
+			if (notifyCollection != null) {
+				notifyCollection.CollectionChanged += (sender, args) => {
+					if (args.NewItems != null)
+						foreach (var newItem in args.NewItems)
+							picker.Items.Add ((newItem ?? "").ToString ());
+
+					if (args.OldItems != null)
+						foreach (var oldItem in args.OldItems)
+							picker.Items.Remove ((oldItem ?? "").ToString ());
+				};
+			}
+
+			var enumerable = newValue as IEnumerable;
+			if (enumerable == null)
+				return;
+
+			picker.Items.Clear ();
+
+			foreach (var item in enumerable)
+				picker.Items.Add ((item ?? "").ToString ());
```

Comment:
Using ToString is probably not what we want here. A mapping function, or a MemberPath should be preferred. And if we do this, there's another control (can't recall which one right now) in Forms that needs the exact same treatment.